modules/product/src/main/java/com/opengamma/strata/product/cms/CmsPeriod.java

```diff
@@ -0,0 +1,1119 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.cms;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.OptionalDouble;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableConstructor;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.BuySell;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.swap.Swap;
+import com.opengamma.strata.product.swap.SwapIndex;
+import com.opengamma.strata.product.swap.type.FixedIborSwapConvention;
+import com.opengamma.strata.product.swap.type.IborRateSwapLegConvention;
+
+/**
+ * A period over which a CMS coupon or CMS caplet/floorlet payoff is paid.
+ * <p>
+ * A single payment period within a CMS leg.
+ * This class specifies the data necessary to calculate the value of the period.
+ * The payment period contains the unique accrual period. 
+ * The value of the period is based on the observed value of {@code SwapIndex}.
+ * <p>
+ * The payment is a CMS coupon, CMS caplet or CMS floorlet. 
+ * If {@code caplet} ({@code floorlet}) is not null, the payment is a caplet (floorlet). 
+ * If both of {@code caplet} and {@code floorlet} are null, this class represents a CMS coupon payment.
+ * Thus at least one of the fields must be null.
+ */
```

Comment:
Add the description of the pay-off. For example:
The pay-offs are, for a SwapIndex on the fixingDate of 'S' and an year fraction 'a'
CMS Coupon: a * S
CMS Caplet: a * (S-K)^+  ; K=caplet
CMS Floorlet: a * (K-S)^+  ; K=floorlet